File: typescript/src/shapes/rectangle.ts
// interface Rectangle {
//     width: number,
//     height: number,
//     computeArea: () => number
// }

// function newRectangle(width: number, height: number): Rectangle {
//     return {
//         width,
//         height,

//         computeArea: function (): number {
//             return width * height
//         }
//     }
// }

// export { Rectangle, newRectangle }

// interface Shape {
//     getArea(): number;
//     draw(): void;
// }

import { Shape } from '../shapes/shape';

class Rectangle implements Shape {
    private height: number;
    private width: number;

    constructor(height: number, width: number) {
        this.height = height;
        this.width = width;
    }

    getArea(): number {
        return this.height * this.width;
    }

    draw(): void {
        console.log(`Shape printed\nIts area is ${this.getArea()}`);
    }
}

export { Rectangle }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Rectangle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Rectangle implements Shape {
    // public double height;
    // public double width;
    
    // public Rectangle(double height, double width){
    //     this.height = height;
    //     this.width = width;
    // }

    // public double getArea() {
    //     return height * width;
    // }

    private double height;
    private double width;

    public Rectangle(double height, double width){
        this.height = height;
        this.width = width;
        
    }

    @Override 
    public double getArea(){
        return height * width;
    }

    @Override
    public void draw(){
        System.out.println("Shape printed\n" + "Its area is " + getArea());
    }
}
